app/src/main/java/org/mozilla/fenix/settings/CustomizationFragment.kt

```diff
@@ -147,26 +147,22 @@ class CustomizationFragment : PreferenceFragmentCompat() {
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_recent_tabs).apply {
-            isVisible = FeatureFlags.showRecentTabsFeature
-            isChecked = context.settings().showRecentTabsFeature
+            isVisible = context.components.analytics.features.homeScreen.isRecentlyTabsActive()
             onPreferenceChangeListener = CustomizeHomeMetricsUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_recent_bookmarks).apply {
-            isVisible = FeatureFlags.recentBookmarksFeature
-            isChecked = context.settings().showRecentBookmarksFeature
+            isVisible = context.components.analytics.features.homeScreen.isRecentlySavedActive()
             onPreferenceChangeListener = CustomizeHomeMetricsUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_pocket_homescreen_recommendations).apply {
-            isVisible = false
-            isChecked = context.settings().pocketRecommendations
+            isVisible = context.components.analytics.features.homeScreen.isPocketRecommendationsActive()
             onPreferenceChangeListener = CustomizeHomeMetricsUpdater()
         }
 
         requirePreference<SwitchPreference>(R.string.pref_key_history_metadata_feature).apply {
-            isVisible = FeatureFlags.historyMetadataUIFeature
-            isChecked = context.settings().historyMetadataUIFeature
```

Comment:
For all of these prefs, I thought:

1. that the `FeatureFlags` were toggling the visibility
2. that Nimbus was toggling the default behaviour.